extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/VertxHttpRecorder.java

```diff
@@ -385,12 +390,16 @@ private static void createPemKeyCertOptions(Path certFile, Path keyFile,
 
     private static HttpServerOptions createHttpServerOptions(HttpConfiguration httpConfiguration,
             LaunchMode launchMode) {
-        // TODO other config properties
-        HttpServerOptions options = new HttpServerOptions();
-        options.setHost(httpConfiguration.host);
-        options.setPort(httpConfiguration.determinePort(launchMode));
-        options.setMaxHeaderSize(httpConfiguration.limits.maxHeaderSize.asBigInteger().intValueExact());
-        return options;
+        if (httpConfiguration.portEnabled) {
+            // TODO other config properties
+            HttpServerOptions options = new HttpServerOptions();
```

Comment:
You still need to create these options if you are going to start the server and send a redirect, otherwise the user can't control the port. The header limits are also important.